src/TrayIcon.cpp

```diff
@@ -0,0 +1,31 @@
+#include "TrayIcon.hpp"
+
+
+TrayIcon::TrayIcon(MainWindow* window, Glib::RefPtr<Gtk::Builder> const& refBuilder)
+{
+    AppIndicator* trayIcon = 
+        app_indicator_new("whatsapp-for-linux", "whatsapp-for-linux", APP_INDICATOR_CATEGORY_COMMUNICATIONS);
+    Gtk::Menu* menu = nullptr;
+    refBuilder->get_widget("indicator_menu", menu);
+    
+    Gtk::MenuItem* quitMenuItem = nullptr;
+    refBuilder->get_widget("quit_indmenu_item", quitMenuItem);
+    quitMenuItem->signal_activate().connect(sigc::mem_fun(this, &TrayIcon::onQuit));
+    
+    Gtk::MenuItem* openMenuItem = nullptr;
+    refBuilder->get_widget("open_indmenu_item", openMenuItem);
+    openMenuItem->signal_activate().connect(sigc::bind(sigc::mem_fun(this, &TrayIcon::onOpen), window));
+
+    app_indicator_set_status(trayIcon, APP_INDICATOR_STATUS_ACTIVE);
```

Comment:
Also, the tray is always displayed even the menuitem is not active.

Reply:
I think it should be. How to do it?

Reply:
Well, we can hold `TrayIcon` with a unique_ptr in MainWindow and initialize/reset when we need it.

Reply:
`APP_INDICATOR_STATUS_PASSIVE` hide tray icon. We can use it.

Reply:
I added function `set_active` for class TrayIcon for show/hide tray icon. And move initialize object TrayIcon to class MainWindow. But now i can't add object TrayIcon as variable of class. When i `#include TrayIcon.hpp` in `MainWindow.hpp`, i get error
```
~/.../build/debug >>> make
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sergey/Загрузки/whatsapp-for-linux/build/debug
Scanning dependencies of target whatsapp-for-linux
[ 12%] Building CXX object CMakeFiles/whatsapp-for-linux.dir/src/main.cpp.o
In file included from /home/sergey/Загрузки/whatsapp-for-linux/src/MainWindow.hpp:4,
                 from /home/sergey/Загрузки/whatsapp-for-linux/src/main.cpp:1:
/home/sergey/Загрузки/whatsapp-for-linux/src/TrayIcon.hpp:11:28: ошибка: expected «)» before «*» token
   11 |         TrayIcon(MainWindow* window, Glib::RefPtr<Gtk::Builder> const& refBuilder);
      |                 ~          ^
      |                            )
/home/sergey/Загрузки/whatsapp-for-linux/src/TrayIcon.hpp:17:21: ошибка: «MainWindow» не был декларирован
   17 |         void onOpen(MainWindow* window);
      |                     ^~~~~~~~~~
make[2]: *** [CMakeFiles/whatsapp-for-linux.dir/build.make:86: CMakeFiles/whatsapp-for-linux.dir/src/main.cpp.o] Ошибка 1
make[1]: *** [CMakeFiles/Makefile2:95: CMakeFiles/whatsapp-for-linux.dir/all] Ошибка 2
make: *** [Makefile:149: all] Ошибка 2
```

Reply:
Ooops, we created a circular dependency. It's okay to have it like before but we need another solution then.

Reply:
Or you can pass Gtk:Window instead of MainWindow and include only gtkmm/windo w in tray class

Reply:
Oooh, it's a good idea!